sliceline/slicefinder.py

```diff
@@ -109,6 +110,9 @@ def _check_params(self):
 
         if self.min_sup < 0:
             raise ValueError(f"Invalid 'min_sup' parameter: {self.min_sup}")
+        
+        if 0 < self.min_sup < 1:
+            self.min_sup = int(self.min_sup * n_samples)
```

Comment:
IMO it is better only to check parameters here and not modify anything. I would check that if `min_sup` is a float it is between 0 and 1 and is an integer otherwise.

Then in the `fit` function, we can adapt `self.min_sup` accordingly